### ISSUE TITLE
training code

### ISSUE THREAD

**Issue by an1018** [fh2019ustc/DocTr#14]
Hi，thanks for your great work, and when will you release the training code?

**Comment by fh2019ustc** [comment]
Hi, thanks for your attention to our work. We will release the training code after the acceptance of our work [DocScanner](https://arxiv.org/pdf/2110.14968v2.pdf).

**Comment by an1018** [comment]
Thanks for your reply, could you tell us your training environment（such as, the number and  model of GPU）、the training time of geometric unwarping transformer and  illumination correction transformer

**Comment by fh2019ustc** [comment]
For geometric unwarping, we use 4 GPUs for training. The training takes about 3 days. 
For illumination correction, we use 2 GPUs for training. The training takes about 1 day.
In fact, we do not conduct hyper-parameter tuning experiments on the batch size, learning rate, and number of GPU.

**Comment by an1018** [comment]
Thanks for your detailed explanation, and  the training GPUs of DocScanner is NVIDIA RTX 2080 Ti GPUs and  NVIDIA GTX 1080 Ti GPU, which one is used in DocTr?

**Comment by fh2019ustc** [comment]
Hi, for DocTr we use 1080 Ti GPUs. In fact, based on our experience, the category of GPU seems not to affect the performance of our method.

**Comment by an1018** [comment]
When writing the training code,I have some confusion. 
1）Before training the GeoTr module, the background needs to be removed. Is it handled by the pre-trained model of the Segmentation module?  
![image](https://user-images.githubusercontent.com/19808900/200591309-4420f9e2-422a-41cb-af08-02328afb9ddf.png)

2）And after removing the background，the result looks like the image on the right?
![image](https://user-images.githubusercontent.com/19808900/200591983-7e4e226a-d989-46f2-8bbf-95ee21116b49.png)

3）But in DocScanner, is ground truth mask the result of document localization module? If yes, Why does it say groud truth?

![image](https://user-images.githubusercontent.com/19808900/200593501-07dacff7-1103-4e8e-bfd7-4f3bf2d051b2.png)

**Comment by fh2019ustc** [comment]
Thanks for your attention to our work.
1. To train the segmentation module,  we remove the noisy backgrounds using the GT masks rather than the pre-trained segmentation module. This is the same for our DocTr and DocScanner.
2. You can also upsample the mask to the original resolution as the input image and then multiply them at the original resolution.
3. See the A1.

Hope this helps.

**Comment by an1018** [comment]
Is there any reference code? And what does GT masks represent in the doc3d dataset?
![image](https://user-images.githubusercontent.com/19808900/200717437-5e54c2b3-cb8f-420c-a976-208ecde759c5.png)

**Comment by fh2019ustc** [comment]
In fact, it is easy to extract the GT mask of the document image from other annotations. For example, in UV map, the values of the background region are 0.

**Comment by an1018** [comment]
@fh2019ustc I've written the training code, but the model does not converge. I'vd send the code to your email（haof@mail.ustc.edu.cn）, could you look at the code？Thanks very much.

**Comment by Aiden0609** [comment]
@an1018 So, have you reproduced it successfully with your own training code?

**Comment by minhduc01168** [comment]
@an1018 So have you successfully written your own training code?

**Comment by zhaolitc** [comment]
> In fact, it is easy to extract the GT mask of the document image from other annotations. For example, in UV map, the values of the background region are 0.

In other types of the Doc3D, take Backward Mapping for example, the values of the background region are also 0 value? When Loss function compare the Ground Truth backward map "bm_gt" and the output "bm_tr" of DocTr, do I need to remove the backgound in  bm_gt? Thank you!
